src/server.c

```diff
@@ -1439,7 +1439,10 @@ void beforeSleep(struct aeEventLoop *eventLoop) {
 
     /* Check if there are clients unblocked by modules that implement
      * blocking commands. */
-    if (moduleCount()) moduleHandleBlockedClients();
+    if (moduleCount()) {
+        moduleFireServerEvent(REDISMODULE_EVENT_BEFORE_SLEEP, 0, NULL);
```

Comment:
maybe it should be the very last thing before going to sleep (before unlocking the GIL)?
i imagine there's some cases where it makes sense to do that earlier (e.g. before `flushAppendOnlyFile`), and some cases where it makes sense to do that after (e.g. after `handleClientsBlockedOnKeys`.

Reply:
I added here to give module a chance to unblock clients in the callback. Next function is moduleHandleBlockedClients() here. This way it can flush out clients before going to sleep. 

I can't wrap my head around handleClientsBlockedOnKeys :) why do you think it makes sense to trigger callback after it?

Reply:
i don't know.. this is a generic callback, so i don't know what it'll be used for.
it's just a thought... not an argument against what you did.